nova/virt/vmwareapi/vmops.py

```diff
@@ -1552,6 +1558,49 @@ def _clean_up_after_special_spawning(self, context, instance_memory_mb,
                 placement_client.set_inventory_for_provider(context, rp.uuid,
                                                             inv_data)
 
+    def _stack_vm_to_host_if_needed(self, instance):
+        """Selects the fullest host that can fit the VM.
+
+        This runs only if DRS is disabled and the configuration
+        CONF.vmware.drs_disabled_stack_vms is set to True
+
+        :returns: HostSystem mo-ref or None
+        """
+        if not CONF.vmware.drs_disabled_stack_vms:
+            return None
+        if cluster_util.is_drs_enabled(self._session, self._cluster):
+            return None
+        LOG.info("DRS is disabled and stacking VMs was enabled.")
+
+        hosts_free = []
+        hosts_full = []
+        for host_ref, host in vm_util.get_stats_from_cluster_per_host(
```

Comment:
Same here: `host` is a quite confusing name as one (I) would rather expect a moref/name in there than info.

Why do we call `vm_util.get_stats_from_cluster_per_host()` here and can't re-use VCState - maybe `VCState.get_host_stats(refresh=True)` if we need the most-current data. I guess we then also need the merged cluster information - is that the problem?